--- /usr/lib/python2.7/site-packages/ansible/module_utils/facts.py
+++ /usr/lib/python2.7/site-packages/ansible/module_utils/facts.py
@@ -280,7 +280,7 @@
         if data:
             self.facts['cmdline'] = {}
             try:
-                for piece in shlex.split(data):
+                for piece in shlex.split(data, posix=False):
                     item = piece.split('=', 1)
                     if len(item) == 1:
                         self.facts['cmdline'][item[0]] = True
