--- /usr/lib/python2.7/site-packages/ansible/playbook/role/requirement.py
+++ /usr/lib/python2.7/site-packages/ansible/playbook/role/requirement.py
@@ -83,9 +83,8 @@
         #   'version': 'v1.0',
         #   'name': 'repo'
         # }
+        display.deprecated("The comma separated role spec format, use the yaml/explicit format instead. Line that trigger this: %s" % role_spec)
 
-        display.deprecated("The comma separated role spec format, use the yaml/explicit format instead.")
-
         default_role_versions = dict(git='master', hg='tip')
 
         role_spec = role_spec.strip()
@@ -145,8 +144,13 @@
             return dict(name=name, src=src, scm=scm, version=version)
 
         if 'role' in role:
-            # Old style: {role: "galaxy.role,version,name", other_vars: "here" }
-            role = RoleRequirement.role_spec_parse(role['role'])
+            name = role['role']
+            if ',' in name:
+                # Old style: {role: "galaxy.role,version,name", other_vars: "here" }
+                role = RoleRequirement.role_spec_parse(role['role'])
+            else:
+                del role['role']
+                role['name'] = name
         else:
             role = role.copy()
